fs/xfstests.py

```diff
@@ -40,6 +42,79 @@ class Xfstests(Test):
 
     :avocado: tags=fs,privileged
     """
+    @staticmethod
+    def get_size_alignval():
+        """
+        Return the size align restriction based on platform
+        """
+        if 'Hash' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0'):
+            def_align = 16 * 1024 * 1024
+        else:
+            def_align = 2 * 1024 * 1024
+        return def_align
+
+    def get_half_region_size(self, region):
+        size_align = self.get_size_alignval()
+        region_size = self.plib.run_ndctl_list_val(self.plib.run_ndctl_list(
+            '-r %s' % region)[0], 'size')
+
+        namespace_size = region_size // 2
+        namespace_size = (namespace_size // size_align) * size_align
+        return namespace_size
+
+    @avocado.fail_on(pmem.PMemException)
+    def setup_nvdimm(self):
+        logflag = self.params.get('logdev', default=False)
+
+        self.plib = pmem.PMem()
+        self.plib.enable_region()
+        regions = sorted(self.plib.run_ndctl_list('-R'),
+                         key=lambda i: i['size'], reverse=True)
+        if not regions:
+            self.cancel("Nvdimm test with no region support")
+
+        if logflag and not len(regions) > 1:
+            self.cancel("Nvdimm test with log flag need two region support")
+
+        regions = sorted(self.plib.run_ndctl_list('-R'),
+                         key=lambda i: i['size'], reverse=True)
+        region = self.plib.run_ndctl_list_val(regions[0], 'dev')
+        if self.plib.is_region_legacy(region):
+            self.cancel("Not supported with legacy regions")
+
+        if logflag:
+            region_ldev = self.plib.run_ndctl_list_val(regions[1], 'dev')
+            if self.plib.is_region_legacy(region_ldev):
+                self.cancel("Not supported with legacy regions")
+
+        self.plib.destroy_namespace(region=region, force=True)
+        namespace_size = self.get_half_region_size(region)
+        self.plib.create_namespace(region=region, size=namespace_size)
+        self.plib.create_namespace(region=region, size=namespace_size)
+        namespaces = self.plib.run_ndctl_list('-N -r %s' % region)
+        pmem_dev = self.plib.run_ndctl_list_val(namespaces[0], 'blockdev')
+        self.test_dev = "/dev/%s" % pmem_dev
+        pmem_dev = self.plib.run_ndctl_list_val(namespaces[1], 'blockdev')
+        self.scratch_dev = "/dev/%s" % pmem_dev
+        self.devices.extend([self.test_dev, self.scratch_dev])
+
+        if logflag:
+            # log device to be created in sector mode
+            self.plib.destroy_namespace(region=region_ldev, force=True)
+            namespace_size = self.get_half_region_size(region=region_ldev)
+            self.plib.create_namespace(region=region_ldev, mode='sector',
+                                       sector_size='512', size=namespace_size)
```

Comment:
Logdev for xfs filesystem should be less than 2G. 
```
# mkfs.xfs -f -b size=64k -s size=512 -m reflink=0 -l logdev=/dev/pmem1s /dev/pmem0
log size 53270740992 bytes too large, maximum size is 2136997888 bytes
...
```

Reply:
updated with 
```
+            namespace_size = self.get_half_region_size(region=region_ldev)
+            # XFS restrict max log size to 2136997888
+            namespace_size = max(namespace_size, 2136997888)

```

Reply:
It should be `min(namespace_size, 2136997888)`